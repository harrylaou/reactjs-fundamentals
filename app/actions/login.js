export const LOGIN_USER_OK = 'LOGIN_USER_OK'
export const LOGOUT_USER = 'LOGOUT_USER'

export const loginUser = () => ({
  type: LOGIN_USER_OK
})

export const logoutUser = () => ({
  type: LOGOUT_USER
})