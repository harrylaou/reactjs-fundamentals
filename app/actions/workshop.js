export const RECEIVE_WORKSHOP = 'RECEIVE_WORKSHOP'


export const receiveWorkshop = workshop => ({
  type: RECEIVE_WORKSHOP,
  workshop
})