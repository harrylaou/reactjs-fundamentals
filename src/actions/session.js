export const LOG_OUT = 'LOG_OUT'
export const LOGGED_IN = 'LOGGED_IN'

export const logout = () => ({
  type: LOG_OUT
})

export const loggedIn = token => ({
  type: LOGGED_IN,
  token
})
