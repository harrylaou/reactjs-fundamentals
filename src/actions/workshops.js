export const RECEIVE_WORKSHOPS = 'RECEIVE_WORKSHOPS'

export const receiveWorkshops = workshops => ({
  type: RECEIVE_WORKSHOPS,
  workshops
})
