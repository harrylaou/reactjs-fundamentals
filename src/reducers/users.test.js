import users from './users'
import {RECEIVE_USERS} from '../actions/users'

describe('#Reducer: Users', () => {
  it('Return default state',() => {
    expect(users(undefined, { type: 'default_action' })).toEqual([])
    expect(users(undefined, { type: 'default_action' })).toMatchSnapshot()
  })

  it('Return state',() => {
    expect(users(["data"], { type: 'another_action' })).toEqual(["data"])
    expect(users(["data"], { type: 'another_action' })).toMatchSnapshot()
  })

  it(`When Action ${RECEIVE_USERS} return state`,() => {
    expect(users(["data"], { type: RECEIVE_USERS, users: ["Eduard"] })).toEqual(["Eduard"])
    expect(users(["data"], { type: RECEIVE_USERS, users: ["Eduard"] })).toMatchSnapshot()
  })
})