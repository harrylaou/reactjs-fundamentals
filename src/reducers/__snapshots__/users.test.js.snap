// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Reducer: Users Return default state 1`] = `Array []`;

exports[`#Reducer: Users Return state 1`] = `
Array [
  "data",
]
`;

exports[`#Reducer: Users When Action RECEIVE_USERS return state 1`] = `
Array [
  "Eduard",
]
`;
